### PR TITLE
Seed selector: support gemini test

### PR DESCRIPTION
**Pull request by juliayakovlev**
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/tester.py [hunk 1]
```diff
@@ -233,16 +233,23 @@ def setUp(self):
         self.params['update_db_packages'] = download_dir_from_cloud(update_db_packages)
 
         self.init_resources()
-        self.set_seeds()
 
-        # Init seed nodes
-        self.db_cluster.wait_for_init(node_list=self.db_cluster.seed_nodes)
+        def init_nodes(clus):
```
Comment (bentsi) [inline sdcm/tester.py]: `clus` -> `_cluster`

File: sdcm/tester.py [hunk 2]
```diff
@@ -233,16 +233,23 @@ def setUp(self):
...
+            clus.set_seeds(seeds_selector=self.params.get('seeds_selector'),
```
Comment (bentsi) [inline sdcm/tester.py]: Why do get `seeds_selector`, `seeds_num` as arguments? `params` object is available on cluster objects

File: sdcm/tester.py (hunk 1 above)
Comment (juliayakovlev) [reply sdcm/tester.py]: I called it first "cluster" and commit failed with :
```
hookid: pylint

************* Module sdcm.tester
W:237,23: Redefining name 'cluster' from outer scope (line 43) (redefined-outer-name)

```

File: sdcm/tester.py [hunk 3]
```diff
@@ -233,16 +233,23 @@ def setUp(self):
...
+                           seeds_num=self.params.get('seeds_num'))
 
-        # Init non-seed nodes
-        self.db_cluster.wait_for_init(node_list=self.db_cluster.non_seed_nodes)
+            # Init seed nodes
+            clus.wait_for_init(node_list=clus.seed_nodes)
 
+            # Init non-seed nodes
+            if clus.non_seed_nodes:
+                clus.wait_for_init(node_list=clus.non_seed_nodes)
```
[the remark this answers is not in the document]
Comment (bentsi) [reply sdcm/tester.py]: Right, I would create a static method on tester class or find a fancier solution like hiding it in `wait_for_init` for example

[the remark this answers is not in the document]
Comment (juliayakovlev) [reply sdcm/tester.py]: @bentsi  @aleksbykov 
May be move it to the cluster?

[the remark this answers is not in the document]
Comment (juliayakovlev) [reply sdcm/tester.py]: Fixed

File: sdcm/tester.py (hunk 2 above)
Comment (juliayakovlev) [reply sdcm/tester.py]: The function has been removed

File: sdcm/cluster.py [hunk 4]
```diff
@@ -2925,7 +2952,8 @@ def wait_for_init(self, node_list=None, verbose=False, timeout=None):  # pylint:
         :param timeout: timeout in minutes to wait for init to be finished
         :return:
         """
-        node_list = node_list or self.nodes  # pylint: disable=no-member
+        # if not node_list, first init seeds (self.seed_nodes), and after non seeds (self.non_seed_nodes)
+        node_list = node_list or self.seed_nodes + self.non_seed_nodes
```
Comment (bentsi) [inline sdcm/cluster.py]: just for protocol, `node_setup` is run in  `wait_for_init_wrap` so the reorder of the node setup will not work here. 
move the logic to setup and create separate method that starts seeds first and the other nodes

**Comment by bentsi** [comment]
not backporting this still not stable. If decided, will need to backport after #1336

**Pull request merged**